### PR TITLE
chore: enabled tests in CI

### PR DESCRIPTION
**Pull request by mauricedesaxe**
I enabled tests in CI because I have now added .env vars so everything should work fine.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,13 +23,21 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
 
+      - name: "Create env file"
+        run: |
+          touch .env
+          echo POLYGON_PRIVATE_KEY=${{ secrets.POLYGON_PRIVATE_KEY }} >> .env
+          echo POLYGON_URL=${{ secrets.POLYGON_URL }} >> .env
+          cat .env
```
Comment (kargakis) [inline .github/workflows/ci.yml:31]: I think you can remove `cat` and `touch` above.

Comment (lazaralex98) [reply .github/workflows/ci.yml:31]: Still doesn't work anyway 🤷🏻‍♂️

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,13 +23,19 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
 
+      - name: "Create env file"
+        run: |
+          echo POLYGON_PRIVATE_KEY=${{ secrets.POLYGON_PRIVATE_KEY }} >> .env
```
Comment (kargakis) [inline .github/workflows/ci.yml:28]: I think we should be running the tests in Mumbai instead of Matic so we don't waste MATIC for every test run.

Comment (kargakis) [reply .github/workflows/ci.yml:28]: Unless we setup the tests to run in a fork so we don't actually execute in the live network.

Comment (lazaralex98) [reply .github/workflows/ci.yml:28]: The tests do run in a fork as off now. But the env variables still seem to not work

**Comment by lazaralex98** [comment]
I figured it out. The reason why my secrets never end up in the env vars is because "Secrets are not passed to workflows that are triggered by a pull request from a fork."

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -28,8 +28,15 @@ jobs:
 
       - name: Build contracts
         run: npx hardhat compile
+        env:
```
Comment (kargakis) [inline .github/workflows/ci.yml:31]: Why do we need to set these secrets during compile?

**Comment by lazaralex98** [comment]
The problem with your earlier suggestion @kargakis of bypassing the use of env vars by simply using a default node url is that this creates a bug when trying to fork since we are forking from a specific block (which may be very old). I don't think we can use a default node url and testing without specifying a block number will result in errors as the token balances of the impersonated accounts can change.

I'll have to do more research on using env vars in CI this weekend.

**Comment by lazaralex98** [comment]
@marceljay can I see how the env vars for tokenizer are set up?

**Comment by kargakis** [comment]
> The problem with your earlier suggestion @kargakis of bypassing the use of env vars by simply using a default node url is that this creates a bug when trying to fork since we are forking from a specific block (which may be very old).

How is a default RPC URL different from setting it in secrets wrt the forked block?

**Comment by lazaralex98** [comment]
> 

Precisely so. Forking only works properly with archival nodes.

**Comment by lazaralex98** [comment]
The secrets never actually get read by the workflow for whatever reason. I tried putting them in my fork and in the ToucanProtocol/example-implementations, as repo secrets and as env secrets. I've tried specified the env in the .yml file. Nothing works...

**Pull request closed without merging**